### ISSUE TITLE
dancing with pip cache on a python image

### ISSUE THREAD

**Issue by mgaitan** [reproducible-containers/buildkit-cache-dance#9]
I am attempting to use `buildkit-cache-dance` to cache pip dependencies in a GitHub Actions workflow but am encountering issues where the cache is not being used. 

My example repo: [mgaitan/pip-docker-cache-dance](https://github.com/mgaitan/pip-docker-cache-dance)

Consider [this commit](https://github.com/mgaitan/pip-docker-cache-dance/commit/d968d0468c4ab7f9230ab189ff87c9615a9dc7aa) where I removed a dependency, while supposely the rest are available in the cache. 

However [the logs indicates](https://github.com/mgaitan/pip-docker-cache-dance/actions/runs/6422637201/job/17439543980#step:7:208) that despite the cache directive, pip dependencies are being downloaded again. 

I'd appreciate any insights or assistance to resolve this issue.

**Comment by henryjw** [comment]
**TLDR; the files are cached but it's possible that the package manage isn't accessing them**

@mgaitan I used your repo to explore how `buildkit-cache-dance` works. After troubleshooting and adding some logs I've concluded that `buildkit-cache-dance` is working as expected. So my guess is that it's something about how you're installing the modules. Here's the run with all my troubleshooting logs: https://github.com/henryjw/pip-docker-cache-dance/actions/runs/7371832005/job/20059582773

![screenshot of cached files](https://github.com/reproducible-containers/buildkit-cache-dance/assets/3443648/e53430b9-740b-44ee-ae29-107b1fe12bbc)


I'm not experienced enough with Python to know why it's not working for you, but for nodejs using the `pnpm` package manager, there are flags that must be set to tell it to give higher priority to the cached files before attempting to download them.
My install script is `pnpm install --prefer-offline --frozen-lockfile`, where `--prefer-offline` tells the package manager to give priority to the existing files.
